src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,161 @@
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            lastNode = new Node<>(null, value, null);
+            firstNode = lastNode;
+        } else {
+            Node<T> newElement = new Node<>(lastNode, value, null);
+            lastNode.next = newElement;
+            lastNode = newElement;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkBounds(index);
+        if (index == 0) {
+            Node<T> current = new Node<>(null, value, firstNode);
+            firstNode.previous = current;
+            firstNode = current;
+        } else {
+            Node<T> current = getNode(index);
+            Node<T> newNode = new Node<>(current.previous, value, current);
+            current.previous.next = newNode;
+            current.previous = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkBounds(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkBounds(index);
+        Node<T> current = getNode(index);
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkBounds(index);
+        Node<T> removedElement = getNode(index);
+        unlink(removedElement);
+        return removedElement.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = firstNode;
+        for (int i = 0; i < size; i++) {
+            if (object == current.value
+                    || object != null && object.equals(current.value)) {
+                unlink(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
+    private void unlink(Node<T> node) {
+        Node<T> nextNode = node.next;
+        Node<T> previousNode = node.previous;
+        if (nextNode == null && previousNode == null) {
+            firstNode = null;
+            lastNode = null;
+        } else if (previousNode == null) {
+            firstNode = nextNode;
+            nextNode.previous = null;
+        } else if (nextNode == null) {
+            lastNode = previousNode;
+            previousNode.next = null;
+        } else {
+            previousNode.next = nextNode;
+            nextNode.previous = previousNode;
+        }
```

Comment:
let's try to simplify this a bit

first check if `nextNode` is null - write `if-else` construction

then check if `previousNode` is null - write another `if-else` construction

let's also set node's value to null in the end

Reply:
thanks for all comments, fixed and changed!)